src/components/VFooter/VFooter.vue

```diff
@@ -0,0 +1,103 @@
+<template>
+  <footer class="flex flex-col gap-10 px-6 py-10 lg:px-10 xl:flex-row xl:gap-8">
+    <!-- Logo and links -->
+    <div
+      v-if="isContentMode"
+      class="flex flex-col gap-10 md:flex-row md:items-center md:justify-between xl:flex-grow"
+    >
+      <VLink href="/" class="text-dark-charcoal">
+        <VBrand class="text-[18px]" />
+      </VLink>
+      <nav>
+        <ul class="grid grid-cols-2 gap-6 text-sm sm:flex sm:flex-row sm:gap-8">
+          <li v-for="page in allPages" :key="page.id">
+            <VLink
+              class="text-dark-charcoal"
+              :href="page.link"
+              show-external-icon
+              >{{ $t(`navigation.${page.id}`) }}</VLink
+            >
+          </li>
+        </ul>
+      </nav>
+    </div>
+
+    <!-- Locale chooser and WordPress affiliation graphic -->
+    <div class="flex flex-col justify-between gap-10 sm:flex-row xl:flex-grow">
+      <div
+        class="flex h-10 w-full w-full items-center justify-center rounded-sm border border-dark-charcoal-20 bg-white sm:w-50"
+      >
+        {{ $t('footer.wip') }}
+      </div>
+      <VLink
+        href="https://wordpress.org"
+        class="text-dark-charcoal hover:no-underline"
+      >
+        <i18n
+          tag="div"
+          path="footer.wordpress-affiliation"
+          class="flex h-full flex-row items-center gap-2 text-sm"
+        >
+          <template #wordpress>
+            <WordPress class="aria-hidden" />
+            <span class="sr-only">WordPress</span>
+          </template>
+        </i18n>
+      </VLink>
+    </div>
+  </footer>
+</template>
+
+<script lang="ts">
+import { defineComponent, computed, PropType } from '@nuxtjs/composition-api'
+
+import usePages from '~/composables/use-pages'
+
+import VLink from '~/components/VLink.vue'
+import VBrand from '~/components/VBrand/VBrand.vue'
+
+import WordPress from '~/assets/wordpress.svg?inline'
+
+/**
+ * The footer is the section displayed at the bottom of a page. It can contain
+ * some branding, links to other pages and an option to change the language.
+ */
+export default defineComponent({
+  name: 'VFooter',
+  components: {
+    VLink,
+    VBrand,
+    WordPress,
+  },
+  props: {
+    /**
+     * whether the footer is being rendered on a content page or an internal
+     * page; This determines whether the Openverse logo and other links are
+     * displayed.
+     */
+    mode: {
+      type: String as PropType<'internal' | 'content' | undefined>,
```

Comment:
When would the mode be undefined? Would it be safer to have only two variants?

Reply:
Using `undefined` means it can infer the type from the current page.

Reply:
Fixed in f35e8b2.